src/foundry/common/utils/helpers.mjs.d.ts

```diff
@@ -256,7 +265,7 @@ export function getType(
  * @param key    - An object property with notation a.b.c
  * @returns An indicator for whether the property exists
  */
-export declare function hasProperty(object: object, key: string): boolean;
+export declare function hasProperty<T extends InputValue, key extends DotNotationKeys<T>>(object: T, key: key): boolean;
```

Comment:
Nit; type parameters in this repo should be PascalCase so it should be `Key extends DotNotationKeys<T>`